tests/61push/06_push_rules_in_sync.pl

```diff
@@ -0,0 +1,107 @@
+use List::Util qw( first );
+
+my $check_for_push_rules = sub {
+   my ( $sync_body ) = @_;
+
+   my $account_data = $sync_body->{account_data}{events};
+
+   my $push_rule_event = first { $_->{type} eq "m.push_rules" } @$account_data;
+
+   assert_json_keys( $push_rule_event->{content}, qw( global ) );
+};
+
+test "Push rules come down in an initial /sync",
+   requires => [ local_user_fixture() ],
+
+   check => sub {
+      my ( $user ) = @_;
+
+      matrix_sync( $user )->on_done( $check_for_push_rules );
```

Comment:
I am slightly dubious on the idea of using `on_done` to side-effect an exception here, because the exception ends up in the wrong place. Better to `then` it here and have the `$check_for_push_rules` closure return a `Future->done()`, so any error goes to the right place.